kernel/of/fdt.c

```diff
@@ -285,7 +411,13 @@ fdt_string_from_offset(struct fdt* fdt, size_t offset)
 const char*
 fdt_node_name(struct fdt_node* node)
 {
+#ifndef FDT_DEBUG_DEBUG_NAMELESS_NODES
     return node->unit_name;
+#else
+    // If the first character is the terminator, replace it with a debug name
```

Comment:
Actually ya

in `kernel.c`

do `printk("/");` before each node

Reply:
Their names don't actually start with a `/` that's just the way the spec refers to some of them. The root node isn't `/` it is actually nameless the spec uses '/` for the sake of brevity not because that's a true prefix. (Also the debugging wasn't meant for the root node, it's to catch any weird behavior in other nodes)

Reply:
Ok ic